src/Mealz/AccountingBundle/Controller/Payment/CashController.php

```diff
@@ -125,21 +123,17 @@ public function paymentFormHandling(Request $request): RedirectResponse
     }
 
     /**
-     * Show transactions for logged in user
-     * Used in routing as an Action.
-     *
-     * @param Request $request request
-     *
-     * @return \Symfony\Component\HttpFoundation\Response
+     * Show transactions for logged in user.
      */
-    public function showTransactionHistory()
+    public function showTransactionHistory(): Response
     {
         $this->denyAccessUnlessGranted('ROLE_USER');
 
         $profile = $this->getUser()->getProfile();
 
         $dateFrom = new DateTime();
         $dateFrom->modify('-28 days');
+        $dateFrom->setTime(0, 0);
```

Comment:
@chetan-thapliyal Here is the place I took care that the correct start time is set before getFullTransactionHistory is called.

Reply:
You can also replace the lines from 134-136 with "new DateTime("-28 days 00:00:00")".